app/services/UserServiceImpl.js

```diff
@@ -29,7 +29,42 @@ module.exports = function(userDao) {
     });
   };
 
+  const signin = function(userData) {
+    return new Promise((resolve, reject) => {
+      _userDao.signin(userData)
+        .then((user) => {
+          comparePassword(userData.password, user.password)
+          .then((res) => {
+            res === true ? resolve(user) : reject(res);
```

Comment:
Changed resolve to pass back true or false result as opposed to the user object. Removed data from reject.